bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestRowResultAdapter.java

```diff
@@ -0,0 +1,167 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class TestRowResultAdapter {
+
+  private static final ByteString ROW_KEY_ONE = ByteString.copyFromUtf8("one");
+  private static final String COL_FAMILY = "cf";
+  private static final ByteString QUAL_ONE = ByteString.copyFromUtf8("q1");
+  private static final ByteString QUAL_TWO = ByteString.copyFromUtf8("q2");
+  private static final List<String> LABELS = ImmutableList.of("label-a", "label-b");
+  private static final ByteString VALUE = ByteString.copyFromUtf8("value");
+
+  private RowResultAdapter underTest = new RowResultAdapter();
+
+  @Test
+  public void testWithSingleCellRow() {
+    RowAdapter.RowBuilder<Result> rowBuilder = underTest.createRowBuilder();
+
+    rowBuilder.startRow(ROW_KEY_ONE);
+    rowBuilder.startCell(
+        COL_FAMILY, QUAL_ONE, 10000L, Collections.<String>emptyList(), VALUE.size());
+    rowBuilder.cellValue(VALUE);
+    rowBuilder.finishCell();
+    rowBuilder.startCell(COL_FAMILY, QUAL_TWO, 20000L, LABELS, -1);
+    rowBuilder.cellValue(VALUE);
+    rowBuilder.finishCell();
+
+    Result expected =
+        Result.create(
+            ImmutableList.<Cell>of(
+                new RowCell(
+                    ROW_KEY_ONE.toByteArray(),
+                    COL_FAMILY.getBytes(),
+                    QUAL_ONE.toByteArray(),
+                    10L,
+                    VALUE.toByteArray()),
+                new RowCell(
+                    ROW_KEY_ONE.toByteArray(),
+                    COL_FAMILY.getBytes(),
+                    QUAL_TWO.toByteArray(),
+                    20L,
+                    VALUE.toByteArray(),
+                    LABELS)));
+    assertResult(expected, rowBuilder.finishRow());
+    assertEquals(ROW_KEY_ONE, underTest.getKey(expected));
+  }
+
+  @Test
+  public void testWhenSplitCell() {
+    ByteString valuePart1 = ByteString.copyFromUtf8("part-1");
+    ByteString valuePart2 = ByteString.copyFromUtf8("part-2");
+    ByteString valuePart3 = ByteString.copyFromUtf8("part-3");
+
+    RowAdapter.RowBuilder<Result> rowBuilder = underTest.createRowBuilder();
+    rowBuilder.startRow(ROW_KEY_ONE);
+    rowBuilder.startCell(
+        COL_FAMILY, QUAL_ONE, 10000L, Collections.<String>emptyList(), valuePart1.size());
+    rowBuilder.cellValue(valuePart1);
+    rowBuilder.cellValue(valuePart2);
+    rowBuilder.cellValue(valuePart3);
+    rowBuilder.finishCell();
+
+    Result actualResult = rowBuilder.finishRow();
+    Assert.assertArrayEquals(
+        valuePart1.concat(valuePart2).concat(valuePart3).toByteArray(),
+        actualResult.getValue(COL_FAMILY.getBytes(), QUAL_ONE.toByteArray()));
+  }
```

Comment:
Since we just identified bugs in the implementation of error handling, can you add tests to make sure it doesnt happen again